### ISSUE TITLE
Update thaliMobile for WiFi and Android

### ISSUE THREAD

**Issue by yaronyg** [thaliproject/Thali_CordovaPlugin#899]
These changes depend on thaliMobileNativeWrapper to be updated with the Android specific changes and for thaliWifiInfrastructure having been updated.
- [x] Implement getAddressPort
- [x] Implement disconnect for Android and return error for WiFi
- [x] Implement extensive changes to peerAvailabilityChanged for Android
- [x] Implement extensive changes to peerAvailabilityChanged for Wifi
- [ ] Implement new discoveryDOS event that comes out of the new peerAvailabilityChanged work

**Comment by artemjackson** [comment]
`getAddressPort` was renamed to `getPeerHostInfo` thaliproject/Thali_CordovaPlugin#929

**Comment by artemjackson** [comment]
@yarong I needed to refresh in my mind how should `disconnect` behaves, so I dived into specs and got a question [1093#issuecomment-248292099](https://github.com/thaliproject/Thali_CordovaPlugin/pull/1093#issuecomment-248292099). Could you please clearify it?

**Comment by artemjackson** [comment]
> Implement disconnect for Android and return error for WiFi

Implemeted here #1120

**Comment by artemjackson** [comment]
> Implement new discoveryDOS event that comes out of the new peerAvailabilityChanged work

Which limits we are going to specify for each of `connectionTypes`(mpcf, bt, tcp) ?

**Comment by artemjackson** [comment]
> Implement new discoveryDOS event that comes out of the new peerAvailabilityChanged work

Implemented here #1129

**Comment by yaronyg** [comment]
@artemjackson The changes are so extensive between the old and new version that just listing them all out turned into an exercise in rewriting the spec. But we already have the spec. The only sane way to handle this is for you to start [here](https://github.com/thaliproject/Thali_CordovaPlugin/blob/22f1a62f23ca1ba0d744a7380893b22f98890724/thali/NextGeneration/thaliMobile.js#L484) and compare what the spec says to do versus what the code actually does. The changes are absolutely everywhere. Just remember, the spec is the truth. Where the code doesn't agree with the spec, the spec is right.

**Comment by yaronyg** [comment]
This is blocked on thaliproject/Thali_CordovaPlugin#902

**Comment by chapko** [comment]
Just some notes about this issue:
- `vNext_yarong_417_812_artemjackson_899_0` - contains platform retrieving changes (utils/platform). Already merged into master
- `vNext_yarong_417_812_artemjackson_899_1` - `disconnect`. Not merged
- `vNext_yarong_417_812_artemjackson_899_2` - discovery DOS event. Not merged.

PR #1129 contains `vNext_yarong_417_812_artemjackson_899_1` and `vNext_yarong_417_812_artemjackson_899_2` branches

**Comment by chapko** [comment]
"Heuristics for marking peers as not available" is not a part of this issue

**Comment by chapko** [comment]
There are 4 branches for this issue:
* `iOS_chapko_899-getPeerHostInfo` - still WIP, notification layer does not work
* `iOS_chapko_899-disconnect` - Artem's commits (`vNext_yarong_417_812_artemjackson_899_1` branch) rebased on top of `iOS` branch
* `iOS_chapko_899-peerAvailabilityChanged` - #1238 PR
* `iOS_chapko_899-discoveryDOS` - rebased Artem's commits from `vNext_yarong_417_812_artemjackson_899_2` branch

**Comment by chapko** [comment]
blocked on #1510

**Comment by chapko** [comment]
"Implement new discoveryDOS event" is now a separate issue #1546.
